client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -125,16 +127,34 @@ public static void setTimeout(Object proxy, int timeout, ChronoUnit timeoutUnit)
         if (Proxy.isProxyClass(proxy.getClass())) {
             Object handler = Proxy.getInvocationHandler(proxy);
             if (handler instanceof HttpClient) {
-                ((HttpClient)handler).setTimeout(timeout, timeoutUnit);
+                ((HttpClient) handler).setTimeout(timeout, timeoutUnit);
             }
         }
     }
 
+    public static void markHeaderAsSensitive(Object proxy, String header) {
+        markHeadersAsSensitive(proxy, singletonList(header));
+    }
+
+    public static void markHeadersAsSensitive(Object proxy, List<String> headers) {
+        if (Proxy.isProxyClass(proxy.getClass())) {
```

Comment:
What do you think about creating a small method that does this check and accepts a lambda as a parameter where we can actually do the mutation of the HttpClient object

With this way we can reuse that method both here and in the setTimeout method.
Something like this:

private static void ifHttpClientDo(Object proxy, Consumer<HttpClient> consumer) {
    if (Proxy.isProxyClass(proxy.getClass()){
Object handler = Proxy.getInvocationHandler(proxy);
if (handler instanceof HttpClient) {
consumer.accept((HttpClient)handler);
}

} 
}

Reply:
Done! Looks much neater now. Thanks!